src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -505,6 +505,26 @@ public SymbolInfo GetSymbolInfo(PositionalPatternClauseSyntax node, Cancellation
             return this.GetSymbolInfoWorker(node, SymbolInfoOptions.DefaultOptions, cancellationToken);
         }
 
+        /// <summary>
+        /// Gets the SymbolInfo for the preprocessing symbol defined in a #define directive trivia, if any.
+        /// </summary>
+        public SymbolInfo GetSymbolInfo(DefineDirectiveTriviaSyntax node)
+        {
+            CheckSyntaxNode(node);
+            var preprocessingSymbol = CreatePreprocessingSymbol(node.Name);
+            return new(preprocessingSymbol);
+        }
+
+        /// <summary>
+        /// Gets the SymbolInfo for the preprocessing symbol defined in an #undef directive trivia, if any.
+        /// </summary>
+        public SymbolInfo GetSymbolInfo(UndefDirectiveTriviaSyntax node)
+        {
+            CheckSyntaxNode(node);
+            var preprocessingSymbol = CreatePreprocessingSymbol(node.Name);
+            return new(preprocessingSymbol);
+        }
```

Comment:
i feel like thse two methods aren't needed.  They could be built off compilation.CreatePreprocessingSymbol and the node.Name of tehse of these pieces of syntax by the caller.